zilliqa/src/api/eth.rs

```diff
@@ -93,6 +93,48 @@ pub fn rpc_module(node: Arc<Mutex<Node>>) -> RpcModule<Arc<Mutex<Node>>> {
     module
 }
 
+// See https://eips.ethereum.org/EIPS/eip-1898
+fn build_errored_response_for_missing_block(
+    request: BlockId,
+    result: Option<Block>,
+) -> Result<Block> {
+    // Block has been found
+    if let Some(block) = result {
+        return Ok(block);
+    }
+
+    // There's no specific error for requests not having block_hash and optional canonical parameter
+    let BlockId::Hash(RpcBlockHash {
+        require_canonical, ..
+    }) = request
+    else {
+        bail!("Unable to find a block with given id!");
+    };
+
+    let require_canonical = require_canonical.unwrap_or_default();
+
+    match require_canonical {
```

Comment:
Doesn't this return the wrong error if I pass `require_canonical: true` and the block isn't found?